rust/src/transformations/cast/ffi.rs

```diff
@@ -64,20 +82,37 @@ pub extern "C" fn opendp_transformations__make_cast_default(
 
 #[no_mangle]
 pub extern "C" fn opendp_transformations__make_cast_inherent(
-    TIA: *const c_char,
+    input_domain: *const AnyDomain,
+    input_metric: *const AnyMetric,
     TOA: *const c_char,
 ) -> FfiResult<*mut AnyTransformation> {
-    let TIA = try_!(Type::try_from(TIA));
+    let input_domain = try_as_ref!(input_domain);
+    let input_metric = try_as_ref!(input_metric);
+    let M = input_metric.type_.clone();
+    let TIA = try_!(input_domain.type_.get_atom());
     let TOA = try_!(Type::try_from(TOA));
 
-    fn monomorphize<TIA, TOA>() -> FfiResult<*mut AnyTransformation>
+    fn monomorphize<M, TIA, TOA>(
+        input_domain: &AnyDomain,
+        input_metric: &AnyMetric,
+    ) -> FfiResult<*mut AnyTransformation>
     where
         TIA: 'static + Clone + CheckAtom,
         TOA: 'static + RoundCast<TIA> + InherentNull + CheckAtom,
+        M: 'static + DatasetMetric,
```

Comment:
This has almost exactly the same pattern as `opendp_transformations__make_cast`, except the `M` where clause is in a different place. It hurts my brain! 😂

Reply:
Ah thanks, I buggered the ordering.